js/data.js

```diff
@@ -6,3 +6,14 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var previousJSON = localStorage.getItem('data');
+
+function beforeUnload(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('data', dataJSON);
+}
+
+window.addEventListener('beforeunload', beforeUnload);
+
+data = JSON.parse(previousJSON);
```

Comment:
You should only do this if there is something stored in localStorage. This will make the application break the first time a user visits it (which is the most important time to work correctly 😬 )